trade-finance-manager-ui/server/controllers/case/cancellation/reason-for-cancelling.controller.ts

```diff
@@ -14,17 +16,34 @@ export type PostReasonForCancellingRequest = CustomExpressRequest<{ params: { _i
  * @param req - The express request
  * @param res - The express response
  */
-export const getReasonForCancelling = (req: GetReasonForCancellingRequest, res: Response) => {
+export const getReasonForCancelling = async (req: GetReasonForCancellingRequest, res: Response) => {
   const { _id } = req.params;
-  const { user } = asUserSession(req.session);
-
-  const reasonForCancellingViewModel: ReasonForCancellingViewModel = {
-    activePrimaryNavigation: PRIMARY_NAVIGATION_KEYS.ALL_DEALS,
-    user,
-    ukefDealId: '0040613574', // TODO: DTFS2-7350 get values from database
-    dealId: _id,
-  };
-  return res.render('case/cancellation/reason-for-cancelling.njk', reasonForCancellingViewModel);
+  const { user, userToken } = asUserSession(req.session);
+  try {
+    const deal = await api.getDeal(_id, userToken);
+
+    if (!deal || 'status' in deal) {
+      return res.redirect('/not-found');
+    }
+
+    if (!canSubmissionTypeBeCancelled(deal.dealSnapshot.submissionType)) {
+      return res.redirect(`/case/${_id}/deal`);
+    }
+
+    const cancellation = await api.getDealCancellation(_id, userToken);
+
```

Comment:
Should we verify `cancellation` object for essential properties?

Reply:
I'm not sure what you mean? Do you mean check that the object returned by `tfm-api` is valid? I can add something like zod validation to this response type but its not done anywhere else in tfm-ui

Reply:
```javascript
if (!cancellation?.reason) {...}
```

I meant something like above, this is quite common across DTFS.